src/grub.sh

```diff
@@ -0,0 +1,110 @@
+# This function list all the available kernels in a VM, local, and remote
```

Comment:
I like the idea of isolate bootloader-specific code... However, I'm not sure if I like how you isolate the bootloader code; notice that you moved many non-specific codes to `grub.sh`. I expected only grub-related things in this part, not all of the other stuff. Additionally, I would prefer a file in `src/bootloader.sh` that manages generic things and specific bootloader code under `src/plugins/bootloader`.